pkg/cleaner/cleaning.go

```diff
@@ -105,18 +108,31 @@ func (c *FileContentObfuscator) ObfuscateFile(inputFile string, outputFile strin
 	// we need to assess whether the file exists already to ensure we don't overwrite existing obfuscated data.
 	// that can happen while obfuscating file names and their paths.
 	// Additionally, the stat check is required because os.O_CREATE will implicitly os.O_TRUNC if a file already exist
+	// Also defining a lock to avoid collisions while creation of files in a multi-threaded environment
+	var mutexLock sync.Mutex
+	mutexLock.Lock()
 	_, err = os.Stat(writePath)
 	if err != nil && !os.IsNotExist(err) {
 		return fmt.Errorf("failed to determine if %s already exists: %w", writePath, err)
 	}
 	if err == nil {
-		return fmt.Errorf("file %s already exists, check whether the obfuscators overwrite each other", writePath)
+		for {
+			fileExt++
+			samplePath := writePath + "." + strconv.Itoa(fileExt)
+			_, err = os.Stat(samplePath)
+			if err != nil && os.IsNotExist(err) {
+				writePath = samplePath
+				break
+			}
+			// return fmt.Errorf("file %s already exists, check whether the obfuscators overwrite each other", writePath)
```

Comment:
please don't comment code, either it's not used then remove it or uncomment it.

Reply:
Addressed